client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -510,7 +516,9 @@ public void write(PendingEvent event) {
                 connection.send(append);
             } catch (ConnectionFailedException e) {
                 log.warn("Failed writing event through writer " + writerId + " due to: ", e);
-                reconnect(); // As the message is inflight, this will perform the retransmission.
+                failConnection(e); // As the message is inflight, this will perform the retransmission.
```

Comment:
Is there a test that checks that this actually can happen? If not, can we add one?

Reply:
Added.